File: assets/conta_palavras_txt.php
<?php
/**
 * Conta palavras com critérios sensatos.
 */

if ($argc<2) exit("\n(CONTA PALAVRAS) falta indicar o path do arquivo\n");
$palavras = preg_split('/[\s\r\n\t\.\?\!\*\(\):,;="]+/', file_get_contents($argv[1]), -1, PREG_SPLIT_NO_EMPTY);
$n=$n2=$n_tudo=0;
foreach($palavras as $palavra) {
  $n_tudo++;
  if (!preg_match('/\d/',$palavra))                $n++;  //nao-digitos
  if (preg_match('/^\p{L}[\-\p{L}]+$/u',$palavra)) $n2++; //nao-digitos e com 2 ou mais letras
}
print "\n --- Contando palavras de $argv[1] ----";
print "\n coisas = $n_tudo\n palavras (nao-digitos) = $n\n palavras com 2 ou mais letras = $n2\n\n";

File: assets/conta_palavras_txt.pl
#!/usr/bin/perl
use warnings;
use strict;
use utf8;

##
## Conta palavras com critérios sensatos. Faz scan na pasta.
##

print STDERR "\n-- Conta palavras de arquivos UTF8 .TXT ou .MD  desta pasta... aguarde.";
my $OUT = '';
my @ls= `ls *txt *.md | grep -v README.md`;  ## falta if argc arquivo especifico  ## bug find . -iregex '.*\.\(md\|txt\)'
chomp @ls;
foreach my $f(@ls) {
        my $n=0; my $n2=0; my $n_tudo=0; my $txt='';
	open(my $fh, '<:encoding(UTF-8)', $f) or die "Could not open file '$f' $!";
	while (my $row = <$fh>) { $txt .= $row; }
	foreach my $palavra( split(/[\s\r\n\t\.\?\!\*\(\):,;="]+/s, $txt) ) {  ## testar
	  $n_tudo++;
	  $n++  if ($palavra !~ m/\d/);   # nao-digitos
	  $n2++ if ($palavra =~ m/^\p{L}[\-\p{L}]+$/);  # nao-digitos e com 2 ou mais letras válidas
	  # debug if (length($palavra)>1 && $palavra !~ m/^\p{L}[\-\p{L}]+$/) {$OUT.= "\n\t\t BUG COM $palavra";}
	}
	$OUT .= "\n --- $f ----";
	$OUT .= "\n coisas = $n_tudo\n palavras (nao-digitos) = $n\n palavras com 2 ou mais letras = $n2\n\n";
}
print "\n$OUT\n"; ## falta opcao CSV
print STDERR "\n---\n";




